seacrowd/sea_datasets/m3exam/m3exam.py

```diff
@@ -0,0 +1,177 @@
+"""
+
+_HOMEPAGE = "https://github.com/DAMO-NLP-SG/M3Exam"
+
+_LANGUAGES = ["jav", "tha", "vie"]
+_LANG_MAPPER = {"jav": "javanese", "tha": "thai", "vie": "vietnamese"}
+_LICENSE = Licenses.CC_BY_NC_SA_4_0.value
+
+_LOCAL = False
+_PASSWORD = "12317".encode("utf-8")  # password to unzip dataset after downloading
+_URLS = {
+    _DATASETNAME: "https://drive.google.com/uc?id=1eREETRklmXJLXrNPTyHxQ3RFdPhq_Nes",
+}
+
+_SUPPORTED_TASKS = [Tasks.QUESTION_ANSWERING]
+
+_SOURCE_VERSION = "1.0.0"
+
+_SEACROWD_VERSION = "1.0.0"
+
+
+class M3Exam(datasets.GeneratorBasedBuilder):
+
+    SOURCE_VERSION = datasets.Version(_SOURCE_VERSION)
+    SEACROWD_VERSION = datasets.Version(_SEACROWD_VERSION)
+
+    BUILDER_CONFIGS = [SEACrowdConfig(name=f"{_DATASETNAME}_{lang}_source", version=datasets.Version(_SOURCE_VERSION), description=f"{_DATASETNAME} source schema", schema="source", subset_id=f"{_DATASETNAME}") for lang in _LANGUAGES] + [
+        SEACrowdConfig(
+            name=f"{_DATASETNAME}_{lang}_seacrowd_qa",
+            version=datasets.Version(_SEACROWD_VERSION),
+            description=f"{_DATASETNAME} SEACrowd schema",
+            schema="seacrowd_qa",
+            subset_id=f"{_DATASETNAME}",
+        )
+        for lang in _LANGUAGES
+    ]
+
+    DEFAULT_CONFIG_NAME = f"{_DATASETNAME}_source"
+
+    def _info(self) -> datasets.DatasetInfo:
+
+        if self.config.schema == "source":
+            features = datasets.Features(
+                {
+                    "question_text": datasets.Value("string"),
+                    "background_description": datasets.Sequence(datasets.Value("string")),
+                    "answer_text": datasets.Value("string"),
+                    "options": datasets.Sequence(datasets.Value("string")),
+                    "language": datasets.Value("string"),
+                    "level": datasets.Value("string"),
+                    "subject": datasets.Value("string"),
+                    "subject_category": datasets.Value("string"),
+                    "year": datasets.Value("string"),
+                }
+            )
+        elif self.config.schema == "seacrowd_qa":
+            features = schemas.qa_features
+
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=features,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager: datasets.DownloadManager) -> List[datasets.SplitGenerator]:
+        """Returns SplitGenerators."""
+        urls = _URLS[_DATASETNAME]
+        lang = self.config.name.split("_")[1]
+
+        data_dir = dl_manager.download(urls)
+
+        if not os.path.exists(data_dir + ".zip"):
+            os.rename(data_dir, data_dir + ".zip")
+
+        with zipfile.ZipFile(data_dir + ".zip", "r") as zip_ref:
+            zip_ref.extractall(data_dir, pwd=_PASSWORD)  # unzipping with password
```

Comment:
I got this error while trying to call `load_dataset`, could you check? Let me know if you can't reproduce the issue

```
>>> x = datasets.load_dataset('seacrowd/sea_datasets/m3exam/m3exam.py', name='m3exam_tha_source')
Downloading data: 100%|█████████████████████████████████████████████████████████████████████████| 2.38k/2.38k [00:00<00:00, 1.94MB/s]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/jamesjaya/Library/Python/3.9/lib/python/site-packages/datasets/load.py", line 2152, in load_dataset
    builder_instance.download_and_prepare(
  File "/Users/jamesjaya/Library/Python/3.9/lib/python/site-packages/datasets/builder.py", line 948, in download_and_prepare
    self._download_and_prepare(
  File "/Users/jamesjaya/Library/Python/3.9/lib/python/site-packages/datasets/builder.py", line 1711, in _download_and_prepare
    super()._download_and_prepare(
  File "/Users/jamesjaya/Library/Python/3.9/lib/python/site-packages/datasets/builder.py", line 1021, in _download_and_prepare
    split_generators = self._split_generators(dl_manager, **split_generators_kwargs)
  File "/Users/jamesjaya/.cache/huggingface/modules/datasets_modules/datasets/m3exam/79e5824e0f329a471260f71ab881767300f3ca5a8d06f60c4efd6ff6fef1f5ec/m3exam.py", line 120, in _split_generators
    with zipfile.ZipFile(data_dir + ".zip", "r") as zip_ref:
  File "/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.9/lib/python3.9/zipfile.py", line 1257, in __init__
    self._RealGetContents()
  File "/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.9/lib/python3.9/zipfile.py", line 1324, in _RealGetContents
    raise BadZipFile("File is not a zip file")
zipfile.BadZipFile: File is not a zip file
```

Reply:
Ok, I will work on it this week

Reply:
@jamesjaya I found out the reason of this problem. While using dl_manager.download(urls) instead of downloading dataset it downloads google page.
![001](https://github.com/SEACrowd/seacrowd-datahub/assets/44943993/bacf2ce2-22e9-4fad-811d-18f1f9100671)

I will try do resolve this problem without using additional libraries. But it seems I will need to use gdown library (if I will use gdown library need I create another PR for adding it to requirements.txt?)

Reply:
@jamesjaya done changes, should work now